3741/1/lib/gem_data_reader.rb

```diff
@@ -0,0 +1,67 @@
+require 'nokogiri'
+require 'json'
+require_relative '../lib/gem_data'
+
+# parsing data from web page of gem
+class GemDataReader
+  attr_reader :html, :gem_name
```

Comment:
you never use html outside
it is not part of interface - remove it 

just use local variable in parse_by method

  attr_reader :gem_name  #input param
  attr_reader :gem_data    #output artifact

make the interface like this:
```
gem_data_reader = GemDataReader.new(gem_name)
gem_data_reader.read
gem_data_reader.gem_data
```

This is called command/query separation:
command - you command object to do smth
query - you ask for the state of the object

#read - it is command method
#gem_date - this is query